### PR TITLE
#2012 - Update Callback Logic for Notification Level Callback URL

### PR DESCRIPTION
**Pull request by EvanParish**
# Description

This work is connecting the pieces together for sending notification level callbacks. The app will now send callbacks to the endpoint indicated in the `callback_url` field when making a request by default. If there's nothing included in the `callback_url` field, it will default to the previous behavior.

issue #2012 

## How Has This Been Tested?

- [x] Added new unit tests
- [x] Deployed to dev and sent callback to our internal endpoint

Logs captured from notification callback tests: (formatted for easier reading)
Debug log when successfully sending callback
```
Callback successfully sent for notification 8dd31563-0a11-4a3c-8a1e-2664958a4e2f,
url: https://dev.api.notifications.va.gov/internal/callback-test | status code: 200
```

POST response from internal endpoint
```
2024-10-10T19:50:11 app app INFO None 
"Generic Internal Request: METHOD: POST 
| ROOT_PATH:  
| PATH: /internal/callback-test 
| QUERY_STRING: b'' 
| URL_RULE: /internal/<generic> 
| TRACE_ID: None 
| JSON: {'id': '8dd31563-0a11-4a3c-8a1e-2664958a4e2f', 'reference': None, 'to': 'evan@email', 'status': 'delivered', 'created_at': '2024-10-10T19:50:06.519910Z', 'completed_at': '2024-10-10T19:50:11.274349Z', 'sent_at': '2024-10-10T19:50:07.145803Z', 'notification_type': 'email', 'status_reason': None, 'provider': 'ses', 'provider_payload': None} 
| HEADERS: X-Forwarded-For: xxx.xxx.xxx.xxx, X-Forwarded-Proto: https, X-Forwarded-Port: 443, Host: dev.api.notifications.va.gov, X-Amzn-Trace-Id: Root=1-x-727b03e674cab6xxxxx, Content-Length: 352, User-Agent: python-requests/2.32.3, Accept-Encoding: gzip, deflate, Accept: */*, Content-Type: application/json, X-Enp-Signature: efb1c4d0a9af0df66166d256b9e6f978430207c2d9d6116882859ad40ec07e55" 
[in /app/app/internal/rest.py:53]
```

retrying on 500 response:
```
Task send-notification-delivery-status[7465e57a-4656-4b74-a8c2-abfa4b8621b8] 
retry: Retry in 563s: AutoRetryException(
  'Found HTTPError, autoretrying...', 
  HTTPError('500 Server Error: INTERNAL SERVER ERROR for url: https://dev.api.notifications.va.gov/internal/qa-automation-callback'), 
  ('500 Server Error: INTERNAL SERVER ERROR for url: https://dev.api.notifications.va.gov/internal/qa-automation-callback',))
```

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/pull_request_labels.md#vanotify-labels) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the [style guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/style_guide.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works. [Testing guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/testing_guide.md)
- [x] I have ensured the latest main is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] The ticket was moved into the DEV test column when I began testing this change

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,6 +35,11 @@ var/
 *.manifest
 *.spec
 
+# pyenv
+#   For a library or package, you might want to ignore these files since the code is
+#   intended to run in multiple environments; otherwise, check them in:
+.python-version
```
Comment (EvanParish) [inline .gitignore:41]: This file helps with my local setup, so excluding it from the repo here.

Comment (MackHalliday) [reply .gitignore:41]: Oooo I might add my own file to ignore in the future.

File: tests/app/celery/test_service_callback_tasks.py [hunk 2]
```diff
@@ -556,3 +582,71 @@ def test_create_delivery_status_callback_data_v3(
...
+        'app.celery.service_callback_tasks.generate_callback_signature',
+        return_value='6842b32e800372de4079e20d6e7e753bad182e44f7f3e19a46fd8509889a0014',
+    )
+    mock_delivery_status_from_notification = mocker.patch(
+        'app.celery.service_callback_tasks.send_delivery_status_from_notification.apply_async'
+    )
+
+    check_and_queue_notification_callback_task(notification)
+
+    # mock_create_callback_data.assert_called_once_with(notification, None, {})
```
Comment (MackHalliday) [inline tests/app/celery/test_service_callback_tasks.py:608]: I'm assuming you meant to delete this comments.

Comment (EvanParish) [reply tests/app/celery/test_service_callback_tasks.py]: yes, good catch.

**Review by MackHalliday — DISMISSED** [review]
Approved. Pulled down branch and reviewed new tests and code. Cross references with user story. 

All edge cases appear to be covered.

**Review by k-macmillan — APPROVED** [review]

**Review by cris-oddball — APPROVED** [review]
